daily_post.py

```diff
@@ -5,23 +5,88 @@
 import praw
 import requests
 import time
+import markovify
+from unidecode import unidecode
+import re
+import random
 
 
 from bot import *
 from bs4 import BeautifulSoup
 from jinja2 import Environment, FileSystemLoader
 
+
+STATE_SIZE = 2
+INFO_URL = 'https://github.com/trambelus/UserSim'
+SUB_URL = '/r/User_Simulator'
+SUB = 'SeattleWA'
+
+FOOTER = '\n\n-----\n\n[^^Info](%s) ^^| [^^Subreddit](%s)' % (INFO_URL, SUB_URL)
+
 reddit = praw.Reddit(client_id=CLIENT_ID,
                     client_secret=CLIENT_SECRET,
                     password=PASSWORD,
                     user_agent=USER_AGENT,
                     username=USERNAME)
 
-subreddit = reddit.subreddit('SeattleWA')
+subreddit = reddit.subreddit(SUB)
 
 now = datetime.datetime.now()
 
 
+class PText(markovify.Text):
+
+    def test_sentence_input(self, sentence):
+        emote_pat = re.compile(r"\[.+?\]\(\/.+?\)")
+        reject_pat = re.compile(r"(^')|('$)|\s'|'\s|([\"(\(\)\[\])])|(github.com/trambelus/UserSim)|(/r/User_Simulator)|(~\ [\w\d\-_]{3,20}\ -----)")
+        decoded = unidecode(sentence)
+        filtered_str = re.sub(emote_pat, '', decoded).replace('  ', ' ')
+        if re.search(reject_pat, filtered_str):
+            return False
+        if filtered_str in FOOTER:
+            return False
+        return True
+
+
+def get_history():
+    try:
+        comments = subreddit.comments(limit=1000)
+        if comments is None:
+            return None, None, None
+        c_finished = False
+        while not c_finished:
+            body = []
+            total_sentences = 0
+            try:
+                for c in comments:
+                    if (not c.distinguished) and ((not subreddit) or c.subreddit.display_name == subreddit):
+                        body.append(c.body)
+                        try:
+                            total_sentences += len(markovify.split_into_sentences(c.body))
+                        except Exception:
+                            total_sentences += 1
+                c_finished = True
+            except praw.exceptions.PRAWException as ex:
+                pass
+        num_comments = len(body)
+        sentence_avg = total_sentences / num_comments if num_comments > 0 else 0
+        body = ' '.join(body)
+        return body, num_comments, sentence_avg
+
+    except praw.exceptions.PRAWException:
+        pass
+
+
+def get_markov():
+    (history, num_comments, sentence_avg) = get_history()
+    try:
+        model = PText(history, state_size=STATE_SIZE)
+        model = PText(history, state_size=STATE_SIZE, chain=model.chain)
```

Comment:
Do we need to seed a Markov with the result of a Markov?

Reply:
Nope, this was left in from troubleshooting my stripped down version of the user_sim code. I'll remove it.